src/sage/schemes/riemann_surfaces/riemann_surface.py

```diff
@@ -2208,31 +2224,30 @@ def local_N(ct, rt):
                 )
                 cg = g(cz, cw)
                 cdgdz = dgdz(cz, cg)
-                Delta = delta_z * cdgdz.abs() + (delta_z**2) * M_tilde / (
-                    rho_z * (rho_z - delta_z)
-                )
-                M = Delta
+                M = delta_z * cdgdz.abs() + pf2 * M_tilde
                 N_required = (
                     (M * (self._RR.pi() + 64 / (15 * (expr**2 - 1))) / E_global).log()
                     / (2 * expr.log())
-                ).ceil()
-                Ni = max(Ni, N_required)
+                )
+                if N_required == Infinity:
+                    return 2**max(60, self._prec)
```

Comment:
I have made the change around check if `N_required` is infinite, but I don't think we need to change the output here. In this case it is not that `N_required` hasn't (initialisation) or can't (because circles won't fit which don't contain branch points) be computed yet, it's that it has been computed but it's just very large (in such a way that floating point errors can distinguish between its denominator and 0). The `nNs` condition thus doesn't need changing later.